src/main/java/org/jabref/gui/StateManager.java

```diff
@@ -41,6 +48,7 @@
     private final OptionalObjectProperty<SearchQuery> activeSearchQuery = OptionalObjectProperty.empty();
     private final ObservableMap<BibDatabaseContext, IntegerProperty> searchResultMap = FXCollections.observableHashMap();
     private final OptionalObjectProperty<Node> focusOwner = OptionalObjectProperty.empty();
+    private final UiThreadObservableList<ObjectProperty<Task<?>>> backgroundTasks = new UiThreadObservableList(FXCollections.observableArrayList());
```

Comment:
What's the reason to wrap the tasks around in a `ObjectProperty`? It should also work without this wrapper. If there were problems with the updates, you might need to add an "extractor" to the `observableArrayList` which specifies that the list should update if the underlying data changes (in this case probably the progress of the task).

Reply:
I'm confused. I though EasyBind's combineList takes care of updating the observable when the progress updates? From the doc:

> Turns an observable list of observable values into a single observable value. The resulting observable value is updated when elements are added or removed to or from the list, as well as when element values change.

I had to turn it into a list of ObjectProperty because that's what EasyBind's combineList expects, but I guess it then only registers changes of the ObjectProperty, not the task.

So to update upon progress changes, I need to tell the list that I am interested in the progress by defining an extractor. I found some code online working with extractors, so I think I should be able to implement that.

Reply:
Yes, combineList takes updates into account but for this the elements have to observable themselves. For example, if `ObservableList<ObservableDoubleValue>` is a list containing the `progress` property of each task.

Reply:
Thanks for the hints. This works!
![Screenshot from 2020-05-09 19-37-23](https://user-images.githubusercontent.com/10479048/81480977-4f4f4480-922d-11ea-8f57-658fc11430ed.png)

Reply:
About the ObjectProperty wrapping issue. I tried changing it back just to see what happens, and I still get the same error for the bindings:

> no instance(s) of type variable(s) T exist so that Task<?> conforms to ObservableValue<? extends T>

I think that the combine method only works on lists of observables.
If I create the binding with Bindings.createDoubleBinding, it does not update.

Reply:
So
```java
public Binding<Double> tasksProgressBinding = Bindings.createDoubleBinding(
            backgroundTasks.filter(Task::isRunning).mapToDouble(Task::getProgress).average().orElse(1), backgroundTasks);`
```
doesn't work?

Reply:
Actually, it does (except for some minor mistakes in your code).
I did not put the list name (backgroundTasks) as a second argument. Does that mean that the binding would have worked ONLY on the extractor? So adding and removing does not update the value, but a change in the progress would have? Or why else do we need to pass the list as a second argument?

Reply:
The second argument specifies the observables that the function (the first argument) depends on. That is, every time these observables change, the function is called and the binding is updated with the new value. If you don't specify any observables in the second argument, then the binding is never updated.

Reply:
Oh that makes sense! Thanks for explaining!